src/main/java/net/sf/jabref/gui/prrv/ReferenceRelationship.java

```diff
@@ -0,0 +1,136 @@
+package net.sf.jabref.gui.prrv;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import net.sf.jabref.JabRefGUI;
+import net.sf.jabref.gui.FXDialogs;
+import net.sf.jabref.logic.l10n.Localization;
+import net.sf.jabref.model.entry.BibEntry;
+
+import prefux.data.Edge;
+import prefux.data.Graph;
+
+/**
+ * @author Daniel Brühl
+ */
+public class ReferenceRelationship {
+
+    // List to store the temporary data for graph visualisation
+    private List<BibEntry> pureEntryList = JabRefGUI.getMainFrame().getCurrentBasePanel().getBibDatabaseContext().getDatabase().getEntries();
+    public List<prefux.data.Node> entryNodeList = new ArrayList<>();
+    public List<prefux.data.Node> externalNodeList = new ArrayList<>();
+    private List<prefux.data.Edge> currentEdgeList = new ArrayList<>();
+
+    /**
+     * Generate necessary node for prefux graph reader.
+     * For every BibTex entry and reference to external papers generate one node
+     */
+    public void parseBibTexForReferences(Graph g) {
+
+        if (JabRefGUI.getMainFrame().getCurrentBasePanel().getBibDatabaseContext().getDatabase().getEntries().isEmpty()) {
+            FXDialogs.showInformationDialogAndWait(Localization.lang("No_references_found"), "Please load a database");
+        } else {
+            // Init - Create nodes for bibtex entries
+            for (int sourceID = 0; pureEntryList.size() > sourceID; sourceID++) {
+                if (pureEntryList.get(sourceID).getField("cites").isPresent() && pureEntryList.get(sourceID).getField("bibtexkey").isPresent()) {
+                    // Split cites for citeCount
+                    String[] citeLines = pureEntryList.get(sourceID).getField("cites").get().split(",");
+                    // Create node of bibtex entry
+                    System.out.println(pureEntryList.get(sourceID).getField("bibtexkey").get());
+                    entryNodeList.add(createNode(sourceID, pureEntryList.get(sourceID).getField("bibtexkey").get(), citeLines.length + 1, "I", g));
+                }
+            }
+
+            // Create edges & external nodes
+            for (int sourceID = 0; pureEntryList.size() > sourceID; sourceID++) {
+                if (pureEntryList.get(sourceID).getField("cites").isPresent()) {
+                    // Split cites
+                    String[] citeLines = pureEntryList.get(sourceID).getField("cites").get().split(",");
```

Comment:
If you convert the array to a list you can also filter the list by using a stream and calling the `filter` method to remove all `null` values. This way you need less if statement inside the for loop.